flink/src/main/java/org/apache/iceberg/flink/CatalogLoader.java

```diff
@@ -43,18 +44,18 @@ static CatalogLoader hive(String name, Configuration hadoopConf, String uri, int
 
   class HadoopCatalogLoader implements CatalogLoader {
     private final String catalogName;
-    private final Configuration hadoopConf;
+    private final SerializableConfiguration hadoopConf;
```

Comment:
Nice catch, since the `CatalogLoader` need to be serializable, so all members need to be serializable. Do we need to add several unit tests to address the serializable issue so that we won't break it in future ?   similar to the unit test TestScanTaskSerialization ?

Reply:
`TableLoader`  need the similar unit test.

Reply:
This is already tested by ITCases. It is the only way which must be passed.
But unit test can be added too.

Reply:
What do you mean ITCases?

Reply:
`TestFlinkScanSql`, `TestFlinkTableSink` and `TestFlinkIcebergSink` will cover this too.

Reply:
I'm not an expert on how Flink will run in tests, but I think it would be safer not to rely on end-to-end or integration tests to check serialization because there may not be a guarantee that tasks are serialized for those runners. (And this was a problem for Hive testing in the past.)

Reply:
Flink `StreamExecutionEnvironment.execute` will launch a local cluster, the client will connect to this cluster by RPC through binary message.
I think you are right, it is better to add unit test too.